Model/OrderSync/Consumer.php

```diff
@@ -0,0 +1,129 @@
+        Config $configHelper,
+        LoggerInterface $logger,
+        Auth $authHelper,
+        StoreKeeperFailedSyncOrderResourceModel $storeKeeperFailedSyncOrderResource,
+        StoreKeeperFailedSyncOrderFactory $storeKeeperFailedSyncOrder,
+        StoreManagerInterface $storeManager,
+        OrderRepositoryInterface $orderRepository
+    ) {
+        $this->ordersHelper = $ordersHelper;
+        $this->logger = $logger;
+        $this->authHelper = $authHelper;
+        $this->configHelper = $configHelper;
+        $this->storeKeeperFailedSyncOrderResource = $storeKeeperFailedSyncOrderResource;
+        $this->storeKeeperFailedSyncOrder = $storeKeeperFailedSyncOrder;
+        $this->storeManager = $storeManager;
+        $this->orderRepository = $orderRepository;
+    }
+
+    /**
+     * Process
+     *
+     * @param string $request
+     * @return void
+     */
+    public function process($request): void
+    {
+        $data = json_decode($request, true);
+
+        try {
+            $orderId = $data['orderId'] ?? null;
+
+            if (is_null($orderId)) {
+                throw new \Exception("Missing order ID");
+            }
+            $order = $this->orderRepository->get($orderId);
+            $storeId = $order->getStoreId();
+
+            $storeKeeperFailedSyncOrder = $this->getStoreKeeperFailedSyncOrder($orderId);
+            try {
+                if ($this->configHelper->isDebugLogs($storeId)) {
+                    $this->logger->info('Processing order: ' . $orderId);
+                }
+                if ($storeKeeperId = $this->ordersHelper->exists($order)) {
+                    $this->ordersHelper->update($order, $storeKeeperId);
+                } else {
+                    $this->ordersHelper->onCreate($order);
+                }
+                if ($storeKeeperFailedSyncOrder->hasData('order_id')) {
+                    $storeKeeperFailedSyncOrder->setIsFailed(0);
+                    $this->storeKeeperFailedSyncOrderResource->save($storeKeeperFailedSyncOrder);
+                }
+            } catch(\Exception $e) {
+                $this->logger->error($e->getMessage());
+                if (!$storeKeeperFailedSyncOrder->hasData('order_id')) {
+                    $storeKeeperFailedSyncOrder->setOrderId((int)$orderId);
+                    $storeKeeperFailedSyncOrder->setIsFailed(1);
+                    $order->setStorekeeperOrderLastSync(time());
+                    $order->setStorekeeperOrderPendingSync(0);
+                    $order->setStorekeeperOrderPendingSyncSkip(true);
+                    $this->orderRepository->save($order);
+                } else {
+                    $storeKeeperFailedSyncOrder->setUpdatedAt(time());
+                }
+                $this->storeKeeperFailedSyncOrderResource->save($storeKeeperFailedSyncOrder);
+            }
+        } catch (\Exception $e) {
+            $this->logger->error("[{$type}] {$entity}({$value}): {$e->getMessage()}");
```

Comment:
```suggestion
            $this->logger->error("[{$type}] {$entity}({$value})", [
                        'exception' => LogHelper::buildReportData($e)
            ]);
            
```
Change also in other places.

In backend we have a function for building usable log context
```php      
    public static function buildReportData(\Throwable $e)
    {
        $data = [];
        if (!is_null($e)) {
            $data = [
                'code' => $e->getCode(),
                'trace' => $e->getTraceAsString(),
                'line' => $e->getLine(),
                'mes' => $e->getMessage(),
                'file' => $e->getFile(),
                'class' => get_class($e),
            ];
            $pe = $e->getPrevious();
            if ($pe) {
                $data['previous'] = self::buildReportData($pe);
            }
        }

        return $data;
    }
```